modules/configuration/README.md

```diff
@@ -195,6 +202,38 @@ public static class DatetimeColumnConfigurationSchema extends ColumnConfiguratio
 Thus, a column can only be one of these (varchar, decimal and datetime) types and will contain the
 type, name and fields specific to it.
 
+### Configuration extension
+
+Allows to add properties into other configurations.
+
+Suppose we have a `security` module and want to add one more authentication component that is located 
+in a different module that depends on `security`.
+
+```java
+@ConfigurationRoot(rootName = "security", type = ConfigurationType.DISTRIBUTED)
+public class SecurityConfigurationSchema {
+    @ConfigValue
+    public AuthenticationConfigurationSchema authentication;
+}
+
+@Config
+public class AuthenticationConfigurationSchema {
+  @Value(hasDefault = true)
+  public final boolean enabled = false;
+}
+```
+
+What we need to do is to subclass the configuration we want to extend.
+
+```java
+@ConfigurationExtension
+public class UserSecurityConfigurationSchema extends SecurityConfigurationSchema {
+  @Value
+  public final String user;
+}
+```
+And the resulting configuration will look as if the field `user` was declared directly in `SecurityConfigurationSchema`.
```

Comment:
Unfortunately, nothing is said about the internal expansion of the configuration.